src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -631,12 +632,13 @@ def forward(
                 `input_features`, the [`WhisperFeatureExtractor`] should be used for extracting the mel features,
                 padding and conversion into a tensor of type `torch.FloatTensor`. See
                 [`~WhisperFeatureExtractor.__call__`]
+            attention_mask (`torch.Tensor`)`, *optional*):
+                Not actually used, pad_tokens are ignored by default. The argument is preserved for compatibility.
```

Comment:
As a user I wouldn't understand that :smile:

Reply:
(neither do I understand it as a reviewer not involved in the initial contribution)

Reply:
We could also just not document it. 

But `attention_mask` is used throughout the library to tell models to skip some `input_ids`.
I think not allowing it is not really respecting the invariant here.

Reply:
We could deprecate the usage directly : 
```suggestion
            attention_mask (`torch.Tensor`)`, *optional*):
                Whisper does not support masking of the `input_features`, this argument is preserved for compatibility, but it is not used. By default the silence in the input log mel spectrogram are ignored. 
```

Reply:
But I agree that having the argument makes a cleaner pipeline call. I am happy either way